llvm/lib/Target/WebAssembly/WebAssemblyPeephole.cpp

```diff
@@ -159,6 +207,9 @@ bool WebAssemblyPeephole::runOnMachineFunction(MachineFunction &MF) {
       case WebAssembly::RETURN:
         Changed |= maybeRewriteToFallthrough(MI, MBB, MF, MFI, MRI, TII);
         break;
+      case WebAssembly::UNREACHABLE:
+        Changed |= eraseDeadCodeAroundUnreachable(MI, MBB);
```

Comment:
Maybe this dead code removal change seems substantial enough as to warrant landing (and testing) separately?

Reply:
You might be right. I put them together, but in different commits, because the dead code removal does essentially what I wanted NoTrapAfterNoreturn to do.

Reply:
+1 on splitting them, but I'm not sure if we should turn `NoTrapAfterNoreturn` off in the first place. The peephole optimization itself can be added regardless of whether we have `NoTrapAfterNoreturn` on or off I guess. (We need to check `noreturn` calls along with `UNREACHABLE`s)